lib.rs

```diff
@@ -310,6 +333,43 @@ impl ThreadPool {
             }
         }
     }
+    
+    /// Block the current thread until all jobs in the pool are completed.
+    /// Once waiting for the pool to complete you can no longer add new jobs, &mut self ensures that.
```

Comment:
The reason I did it in the first place was because without the `mut` many threads can `join` at the same time.

Since the `join` is blocking, the current thread can not send new jobs anyways.
Accessing the pool from many threads requires the `Sync` flag. But I have to reason about that some more because there is an edge case, [see this test](https://github.com/frewsxcv/rust-threadpool/pull/63/commits/78e54dd79ced4613e6fcaa526ff08d1ef6846d18#diff-8c13b5c367b1a0d3dc04da468ea844bdR737).